### PR TITLE
Issue #14631: Updated LITERAL_LITERAL of JavadocTokenTypes to new AST…

### PR DESCRIPTION
**Pull request by RawanWagih**
Issue #14631

test:
```
/**
 * Demonstrates the use of {@literal #compare(Object)} inline tag.
 */
public class Test {
    /**
     * Inline tag example: {@literal #compare(Object)}.
     */
    public void example() {

    }
}
```

output:
```
JAVADOC -> JAVADOC [0:0]
|--TEXT -> /** [0:0]
|--NEWLINE -> \r\n [0:3]
|--LEADING_ASTERISK ->  * [1:0]
|--TEXT ->  Demonstrates the use of  [1:2]
|--JAVADOC_INLINE_TAG -> JAVADOC_INLINE_TAG [1:27]
|   |--JAVADOC_INLINE_TAG_START -> { [1:27]
|   |--LITERAL_LITERAL -> @literal [1:28]
|   |--WS ->   [1:36]
|   |--TEXT -> #compare(Object) [1:37]
|   `--JAVADOC_INLINE_TAG_END -> } [1:54]
|--TEXT ->  inline tag. [1:55]
|--NEWLINE -> \r\n [1:67]
|--LEADING_ASTERISK ->  * [2:0]
|--TEXT -> / [2:2]
|--NEWLINE -> \r\n [2:3]
|--TEXT -> public class Test { [3:0]
|--NEWLINE -> \r\n [3:19]
|--TEXT ->     /** [4:0]
|--NEWLINE -> \r\n [4:7]
|--LEADING_ASTERISK ->      * [5:0]
|--TEXT ->  Inline tag example:  [5:6]
|--JAVADOC_INLINE_TAG -> JAVADOC_INLINE_TAG [5:27]
|   |--JAVADOC_INLINE_TAG_START -> { [5:27]
|   |--LITERAL_LITERAL -> @literal [5:28]
|   |--WS ->   [5:36]
|   |--TEXT -> #compare(Object) [5:37]
|   `--JAVADOC_INLINE_TAG_END -> } [5:54]
|--TEXT -> . [5:55]
|--NEWLINE -> \r\n [5:56]
|--LEADING_ASTERISK ->      * [6:0]
|--TEXT -> / [6:6]
|--NEWLINE -> \r\n [6:7]
|--TEXT ->     public void example() { [7:0]
|--NEWLINE -> \r\n [7:27]
|--TEXT ->         // Example method for testing [8:0]
|--NEWLINE -> \r\n [8:37]
|--TEXT ->     } [9:0]
|--NEWLINE -> \r\n [9:5]
|--TEXT -> } [10:0]
|--NEWLINE -> \r\n [10:1]
`--EOF -> <EOF> [11:0]
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 1]
```diff
@@ -615,12 +615,12 @@ public final class JavadocTokenTypes {
...
      * <pre>
-     * <code> |--JAVADOC_INLINE_TAG[1x0] : [{&#64;literal #compare(Object)}]
-     *        |--JAVADOC_INLINE_TAG_START[1x0] : [{]
-     *        |--LITERAL_LITERAL[1x1] : [@literal]
-     *        |--WS[1x9] : [ ]
-     *        |--TEXT[1x10] : [#compare(Object)]
-     *        |--JAVADOC_INLINE_TAG_END[1x27] : [}]
+     * <code> |--JAVADOC_INLINE_TAG --&gt; {&#64;literal #compare(Object)}
+     *        |--JAVADOC_INLINE_TAG_START --&gt; {
+     *        |--LITERAL_LITERAL --&gt; @literal
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:620]: Should be
```
|--JAVADOC_INLINE_TAG -> JAVADOC_INLINE_TAG [5:27]
   |--JAVADOC_INLINE_TAG_START -> { [5:27]
   |--LITERAL_LITERAL -> @literal [5:28]
   |--WS ->   [5:36]
   |--TEXT -> #compare(Object) [5:37]
   `--JAVADOC_INLINE_TAG_END -> } [5:54]
```

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 2]
```diff
@@ -615,12 +615,13 @@ public final class JavadocTokenTypes {
...
-     *        |--JAVADOC_INLINE_TAG_START[1x0] : [{]
-     *        |--LITERAL_LITERAL[1x1] : [@literal]
-     *        |--WS[1x9] : [ ]
-     *        |--TEXT[1x10] : [#compare(Object)]
-     *        |--JAVADOC_INLINE_TAG_END[1x27] : [}]
+     * <code>
+     *     |--JAVADOC_INLINE_TAG -&gt; JAVADOC_INLINE_TAG [5:27]
+     *        |--JAVADOC_INLINE_TAG_START -&gt; { [5:27]
+     *        |--LITERAL_LITERAL -&gt; @literal [5:28]
+     *        |--WS -&gt;   [5:36]
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:622]: Please remove line and column numbers.
From all lines.

**Review by romani — CHANGES_REQUESTED** [review]
Please reply "done" to all review after you push code to GitHub.

Items:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java (hunk 2 above)
Comment (RawanWagih) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: done

**Comment by RawanWagih** [comment]
Could you please clarify what I should update? Thank you for your guidance!

**Comment by RawanWagih** [comment]
> Ok to merge.
> 
> Please do one more update to be confident on how to do PRs and updates. Thanks a lot in advance.


Could you please clarify what I should update? Thank you for your guidance!

**Comment by romani** [comment]
I meant one update for this issue, different AST in separate PR

**Pull request merged**